metadata-ingestion/src/datahub/ingestion/sink/datahub_rest.py

```diff
@@ -58,6 +58,7 @@ def compute_stats(self) -> None:
 
 
 def _get_urn(record_envelope: RecordEnvelope) -> Optional[str]:
+    assert record_envelope is not None
```

Comment:
the type annotations say that none will never be passed in here, so it feels like the issue is the caller

do you have a stack trace of where this was called with a null argument?

Reply:
Do not have it. Just read the code and found that it directly use the field in a instance. Do not know that python type annotation will hint that it cannot be None.